test/functional/azure_rm_test.rb

```diff
@@ -9,14 +9,81 @@ class ComputeResourcesControllerTest < ActionController::TestCase
+      "disable_password_authentication" => "false",
+      "password" => "testpswd123",
+      "name" => "ervin-golomb.example.com",
+      "image_id" => "OpenLogic:CentOS:7.5:latest",
+      "interfaces_attributes" => {
+        "0" => {"network" => "[\"/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/testrg1/providers/Microsoft.Network/virtualNetworks/first_vnet/subnets/first_subnet\"]", "bridge" => "Dynamic", "name" => "false"}
+      }
+    }.freeze
+
+    test "create vm with password and without custom data" do
+      @test_vnets = mock('vnets')
+      @test_sdk.stubs(:vnets).returns(@test_vnets)
+      @test_v_net = mock('test_v_net')
+      @test_subnets = mock('test_subnets')
+      @test_sdk.stubs(:subnets).returns([@test_subnets])
+      @test_v_net.stubs(:resource_group).returns('testrg1')
+      @test_v_net.stubs(:name).returns('first_vnet')
+      @test_vnets.stubs(:select).returns([@test_v_net])
+      @test_subnet = mock('test_subnet')
+      @test_subnets.stubs(:id).returns('sub_id')
+      @test_nic = mock('test_nic')
+      @test_sdk.stubs(:create_or_update_nic).returns(@test_nic)
+      @test_nic.stubs(:id).returns('nic_id')
+      @test_vm = mock('test_vm')
+      @test_sdk.stubs(:create_or_update_vm).returns(@test_vm)
+      @test_vm_extension = mock('test_vm_extension')
+      @test_sdk.stubs(:create_or_update_vm_extensions).returns(@test_vm_extension)
+      @test_network_profile = mock('test_network_profile')
+      @test_vm.stubs(:network_profile).returns(@test_network_profile)
+      @test_network_interfaces = mock('test_network_interfaces')
+      @test_network_profile.stubs(:network_interfaces).returns([@test_network_interfaces])
+      @test_network_interfaces.stubs(:id).returns('nic_id')
+      @test_vm_nic = mock('test_vm_nic')
+      @test_sdk.stubs(:vm_nic).returns(@test_vm_nic)
+      @test_hardware_profile = mock('test_hardware_profile')
+      @test_vm.stubs(:hardware_profile).returns(@test_hardware_profile)
+      @test_os_profile = mock('test_os_profile')
+      @test_vm.stubs(:os_profile).returns(@test_os_profile)
+      @test_linux_configuration = mock('test_linux_configuration')
+      @test_os_profile.stubs(:linux_configuration).returns(@test_linux_configuration)
+      @test_ssh = mock('test_ssh')
+      @test_linux_configuration.stubs(:ssh).returns(@test_ssh)
+      @test_public_keys = mock('test_public_keys')
+      @test_ssh.stubs(:public_keys).returns(@test_public_keys)
+      @test_storage_profile = mock('test_storage_profile')
+      @test_vm.stubs(:storage_profile).returns(@test_storage_profile)
+      @test_os_disk = mock('test_os_disk')
+      @test_storage_profile.stubs(:os_disk).returns(@test_os_disk)
+      @test_managed_disk = mock('test_managed_disk')
+      @test_os_disk.stubs(:managed_disk).returns(@test_managed_disk)
+      @test_vm.stubs(:name).returns('ervin-golomb.example.com')
+      @test_vm.stubs(:disable_password_authentication).returns(false)
```

Comment:
@ShimShtein Could you give a hint of how do we want to move this common patch into a helper may be?